src/utils/userIP.ts

```diff
@@ -58,14 +66,18 @@ export function processUserIP(req: Request): string | null {
         return parseIP(firstEntry.for);
       } catch (e) {
         logger.warn(
-          `Request received with invalid content in "${HEADER_FORWARDED}" header.`
+          `Request received with invalid content in "${HEADER_FORWARDED}" header.`,
         );
         return null;
       }
     }
     case IPSources.XForwardedFor: {
       logger.trace(`Sourcing User IP from "${HEADER_X_FORWARDED}" header.`);
-      return req.ip ?? null;
+      if (isAPIGatewayProxyEvent(req)) {
+        return req.requestContext?.identity?.sourceIp ?? null;
```

Comment:
If a pod adopts this and isn't using ES6, would if/else statements be more inclusive? I had feedback on my auth PR to avoid from optional chaining, and to amend to if/else statements where possible... wondered what your thoughts are on this?

Reply:
This code is only used by the build command to create the build files. 
Then, build files are slightly different. You will not get any optional chaining on its.
If you have 5 min, build from this branch and look the files inside esm and cjs folders

Reply:
Hi @di-fabs, sorry if you joined the call early and dropped, a few of us were running late.

The conclusion from the call was that in the case of `IPSources.XForwardedFor` we want to return the left-most entry in the `x-forwarded-for` header. This is what `req.ip` does currently for express apps ([docs](https://expressjs.com/en/api.html#:~:text=//%20%3D%3E%20%27example.com%27-,req.ip,-Contains%20the%20remote)), but I guess we don't have that available in APIGateway so we will need to do something similar to how we handle the `forwarded` header.

Reply:
@nickhealGDS @sandy-may 
So I pushed the changes but I'm not sure because without any tech spec..., it's hard to validate it.
1/ New Logic starts with 1 assumption: request type is APIGatewayProxyEvent if request object contains a property "requestContext".
2/ We are looking the X_FORWARDED_FOR header only if request is type APIGatewayProxyEvent.
3/ x-forwarded-for value has this format: "IP, others" (without any spec, I took the format I have seen into MOCK_X_FORWARDED_FOR_IPV4). example:  "198.51.100.13, 2004:db8:cafe::17"
Do you validate these assumptions?